server/BUILD

```diff
@@ -148,20 +148,17 @@ java_binary( # TODO: this target should not be needed if :server-deps-windows ca
     tags = ["maven_coordinates=com.vaticle.typedb:typedb-core-server-bin-deps:{pom_version}"],
 )
 
-java_binary( # TODO: this target should not be needed if :server-deps-prod can depend directly on @maven//:org_rocksdb_rocksdbjni (lib)
+java_binary( # TODO: this target should not be needed if :server-deps-prod can depend directly on @maven//:io_github_speedb_io_speedbjni_2_4_1 (lib)
```

Comment:
This TODO may now be do-able.
Is there a reason we want the `server/lib/{common, prod, dev}` path structure for dependencies?